graph_search.py

```diff
@@ -0,0 +1,115 @@
+class GraphSearch:
+  def dfs(self, target, graph, starting_vertex):
+    info = {
+      "target": target,
+      "found": False,
+      "visited": [],
```

Comment:
What are the consequences of generating a new `visited` array on each iteration?  What happens if you run this on a graph with a cycle?